packages/builder/src/builderStore/store/frontend.js

```diff
@@ -869,7 +877,13 @@ export const getFrontendStore = () => {
           if (!component) {
             return false
           }
-          return patchFn(component, screen)
+
+          // Mutates the fetched component with updates
+          const updated = patchFn(component, screen)
+          // Mutates the component with any required settings updates
+          const migrated = store.actions.components.migrateSettings(component)
```

Comment:
Nice 👌  Good call here by returning a flag to let us know if a migration was required, to let us skip wasted updates.